auth/service.go

```diff
@@ -975,37 +961,6 @@ func (svc service) createDomainPolicyRollback(ctx context.Context, userID, domai
 	return err
 }
```

Comment:
Since DeletePolicy is single operation per user. It difficult to track if any one policy removal fails. It should like single operation for all users. So something like below will keep the removal process in policy and repo in single step.
```suggestion
func (svc service) UnassignUsers(ctx context.Context, token, id string, userIds []string) error {
    pr := PolicyReq{
        Subject:     token,
        SubjectType: UserType,
        SubjectKind: TokenKind,
        Object:      id,
        ObjectType:  DomainType,
        Permission:  SharePermission,
    }
    if err := svc.Authorize(ctx, pr); err != nil {
        return err
    }
    relations := []string{MemberRelation, ViewerRelation, EditorRelation}

    pr.Permission = AdminPermission
    if err := svc.Authorize(ctx, pr); err != nil {
        pr.SubjectKind = UsersKind
        // User is not admin.
        for _, userID := range userIds {
            pr.Subject = userID
            if err := svc.Authorize(ctx, pr); err == nil {
                // Non admin attempts to remove admin.
                return errors.Wrap(svcerr.ErrAuthorization, err)
            }
        }
    } else {
        relations = append(relations, AdministratorRelation)
    }

    if err := svc.removeDomainPolicies(ctx, id, relations, userIds...); err != nil {
        return err
    }
    return nil
}
```